### ISSUE TITLE
CODE 21.11.6 Regression: Does Not Open Specific Calc Spreadsheet

### ISSUE THREAD

**Issue by gohrner** [CollaboraOnline/online#3916]
**Describe the bug**
I have a specific spreadsheet which cannot be opened with COOL 21.11.6. Previously with 6.4.x it worked fine.

Other files (spreadsheets as well as documents) CAN be opened, so the COOL 21.11.6 installation itself basically seems to work.

From a user's point of view, either:

* Some parts of CODE like menu bar, status bar and also sidebar, if expanded, load and display fine, however the document area stays white,
* or on other occasions the screen stays in the loading-artwork, and the progress bar is stuck at 0%.

Unfortunately, in both cases there's nothing interesting in the COOL logs for this file:

```text
wsd-00001-00082 2022-01-02 21:41:32.241483 +0000 [ docbroker_002 ] WRN  Waking up dead poll thread [HttpSynReqPoll], started: false, finished: false| ./net/Socket.hpp:718
wsd-00001-00082 2022-01-02 21:41:32.304258 +0000 [ docbroker_002 ] WRN  Waking up dead poll thread [HttpSynReqPoll], started: false, finished: false| ./net/Socket.hpp:718
wsd-00001-00082 2022-01-02 21:41:36.751748 +0000 [ docbroker_002 ] ERR  Error while handling poll for socket #20 at 0 in docbroker_002: JSON Exception| net/Socket.cpp:468
wsd-00001-00082 2022-01-02 21:42:20.204081 +0000 [ docbroker_002 ] WRN  Ignoring attempted read from 26| ./net/Socket.hpp:1095
^C
```

These log messages look basically the same for files that work.

COOL 21.11.6 in self-build Ubuntu-based Docker container on x64 platform. (Host is Debian Buster.)

Browser is Chromium Version 95.0.4638.69 (Offizieller Build) Built on Ubuntu , running on neon 20.04 (64-Bit).

I cannot publicly attach the problematic file here, and I need to roll back to a working version now, so my testing possibilities are limited.

Maybe someone else experienced similar behaviour?

**Comment by pedropintosilva** [comment]
Thanks @gohrner for reporting it, I see that the file is private but could you perhaps answer these questions? and/or attach a "clean" version without those private bits?:

- How heavy is the file?
- How many cells roughly ?
- Does it have anything out of ordinary? Or function?

**Comment by gohrner** [comment]
The file actually is not really big or complex - in my opinion.

Admittedly, it uses quite a few functions, but has "only" 5 sheets with with about 400 rows and columns up to "W" each plus two much smaller sheets.
It also uses several hundret cell comments, spread across these sheets.

I tried to create a more or less anonymized version and would be willing to provide this to Collabora for debugging, but I still do not want to "publish" it publically on the Internet.

Also I couldn't try if this anonymized version still triggers the issues / opening failure.

**Comment by mmeeks** [comment]
@gohrner it would be wonderful to have a copy of that if we can, can you mail it to Ezinne Martina Nnamani <nnamani.ezinne@collabora.com> and CC me - michael.meeks@collabora.com - thanks for reporting.

**Comment by gohrner** [comment]
> @gohrner it would be wonderful to have a copy of that if we can, can you mail it to Ezinne Martina Nnamani [nnamani.ezinne@collabora.com](mailto:nnamani.ezinne@collabora.com) and CC me - [michael.meeks@collabora.com](mailto:michael.meeks@collabora.com) - thanks for reporting.

Just for the record: I sent a test file yesterday.

**Comment by gohrner** [comment]
> There's bug with similar symptoms: if there's a comment that contains a tab, the control character isn't properly escaped in the JSON, and the result is the same as described:

@thebearon: Interesting catch - how did you find out the cause?

Well possible that the file in question contains some formatting / special characters in comments or cells or whatever.

@eszkadev: In case @thebearon's observation is the full picture, wrapping handling code in try/catch might work around the symptoms, but the "real" solution would probably be to properly escape / quote / encode the input such that no invalid JSON is being generated in the first place, wouldn't it?

I don't know the CODE architecture, but wouldn't it be too late in case an exception already was triggered and/or the JSON cannot be parsed?

The comment should be displayed anyway, also with an included tab or emoji or whatever special character's someone managed to inject there - after all, a tab might be intentionally used for formatting purposes.

**Comment by eszkadev** [comment]
> @eszkadev: In case @thebearon's observation is the full picture, wrapping handling code in try/catch might work around the symptoms, but the "real" solution would probably be to properly escape / quote / encode the input such that no invalid JSON is being generated in the first place, wouldn't it?

real fix is also made, but in the LO core

**Comment by mmeeks** [comment]
Just to say - this was a particularly nasty issue to have; thanks so much for filing the bug & your sample document @gohrner - good stuff.

**Comment by gohrner** [comment]
This does not yet seem to be included in 21.11.1.0 e0cfe36.

What would be the best way for me to get notified once the fixed version is released and I can retry 21.11?

**Comment by Go-rom** [comment]
@gohrner i think that's solved with 21.11.1.1, do you confirm ?

**Comment by gohrner** [comment]
> @gohrner i think that's solved with 21.11.1.1, do you confirm ?

Yes, confirmed. File opens in version

> COOLWSD Version:
21.11.1.1
git hash: 
e487386
LOKit Version:
built on 
a great technology base
Collabora Office 21.06.12.1
git hash: 
45f56ee
>Angeboten von:
>
>Ubuntu 18.04.6 LTS
>
> ef3ec9a0
